### PR TITLE
Remove `Pkg.Types.STDLIBS_BY_VERSION` from Pkg, thus removing the need to constantly keep `STDLIBS_BY_VERSION` up to date

### PR DESCRIPTION
**Pull request by DilumAluthge**
Actually, `Pkg.Types.STDLIBS_BY_VERSION` is still there, but it is now an empty vector. Consumers (e.g. BinaryBuilder) must populate it.

### CODE REVIEW

File: test/new.jl [hunk 1]
```diff
@@ -2790,9 +2794,15 @@ using Pkg.Types: is_stdlib
     @test is_stdlib(pkg_uuid, v"999.999.999")
     @test is_stdlib(pkg_uuid, v"0.7")
     @test is_stdlib(pkg_uuid, nothing)
+
+    empty!(Pkg.Types.STDLIBS_BY_VERSION)
+    @test is_stdlib(networkoptions_uuid)
+    @test_throws Pkg.Types.PkgError is_stdlib(networkoptions_uuid, v"1.6")
```
Comment (staticfloat) [inline test/new.jl:2800]:
```suggestion
    empty!(Pkg.Types.STDLIBS_BY_VERSION)
    
    # Test that we can probe for stdlibs for the current version with no STDLIBS_BY_VERSION,
    # but that we throw a PkgError if we ask for a particular julia version.
    @test is_stdlib(networkoptions_uuid)
    @test_throws Pkg.Types.PkgError is_stdlib(networkoptions_uuid, v"1.6")
```

**Review by staticfloat — APPROVED** [review]
This seems good to me!  We can have a separate package that gets auto-updated outside of the Pkg repository, and use that from BB.  Thanks for doing this Dilum!

**Comment by DilumAluthge** [comment]
@KristofferC Does this look good to you? The idea here is that by removing this list from Pkg, we don't need to constantly make PRs to Pkg to update the list.

**Comment by IanButterworth** [comment]
Bump

**Comment by IanButterworth** [comment]
Given this is lingering, I'll update the stdlibs

**Comment by DilumAluthge** [comment]
Bump @KristofferC

**Comment by DilumAluthge** [comment]
(I can resolve the merge conflicts before merging.)

**Comment by DilumAluthge** [comment]
Bump @KristofferC

**Comment by IanButterworth** [comment]
I'm going to update this and merge it.
The historical stdlib check just failed 5 minutes after a stdlib update PR was merged. It's too much overhead.

AFAIK @KristofferC isn't a user of this interface, so if it's ok by @staticfloat that seems ok to me

**Comment by giordano** [comment]
> Consumers (e.g. BinaryBuilder) must populate it.

How exactly?  Getting a notice about these breaking changes would also be nice.

**Comment by giordano** [comment]
> We can have a separate package that gets auto-updated outside of the Pkg repository, and use that from BB.

Maybe let's merge these PRs ***after*** such packages actually exist instead of breaking downstream packages?

**Comment by IanButterworth** [comment]
Yeah, sorry, did this in the wrong order. Here's the package https://github.com/JuliaPackaging/HistoricalStdlibVersions.jl

Being registered https://github.com/JuliaRegistries/General/pull/71291

**Pull request merged**